tools/nightly-e2e-tests/src/linter.py

```diff
@@ -0,0 +1,93 @@
+# -----------------------------------------------------------
+# Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+# This product includes software developed at Datadog (https://www.datadoghq.com/).
+# Copyright 2019-2020 Datadog, Inc.
+# -----------------------------------------------------------
+
+import contextlib
+from dataclasses import dataclass
+from os.path import abspath
+
+
+@dataclass
+class CodeReference:
+    file_path: str
+    line_no: int
+    line_text: str
+
+
+@contextlib.contextmanager
+def linter_context(code_reference: CodeReference):
+    """
+    Context manager for convenient work with linter's code reference.
+    After context returns, the code reference is reseted to its previous value.
+    """
+    Linter.shared.push_context(code_reference=code_reference)
+    try:
+        yield
+    finally:
+        Linter.shared.pop_context()
+
+
+class NoOpLinter:
+    """
+    Base, no-op linter.
+    """
+
+    shared: 'NoOpLinter'
+
+    def push_context(self, code_reference: CodeReference):
+        pass
+
+    def pop_context(self):
+        pass
+
+    def emit_warning(self, message: str):
+        pass
+
+    def emit_error(self, message: str):
+        pass
+
+
+class Linter(NoOpLinter):
```

Comment:
Is this `XCode` specific ? Are we going to have issues using this script from Android ?

Reply:
The `Linter's` interface is platform agnostic, so I hope this could be _somehow_ integrated with Gradle / Android linter:
```
Linter
- emitWarning(message, file_path, line_number)
- emitError(message, file_path, line_number)
```

The way we integrate it with Xcode build is by printing this to `STDOUT`:
```
<file_path>:<line_number> (warning|error): message
```
Xcode understands this format and renders warnings or errors in appropriate source code locations.